im_library/src/helpers/QueryDescriptor.ts

```diff
@@ -0,0 +1,294 @@
+    if (matchDisplay && matchDisplay.slice(-1) !== ".") matchDisplay += ".";
+    whereDisplays.push(matchDisplay + getDisplayFromWhere(whereItem));
+  }
+
+  return whereDisplays;
+}
+
+export function getDisplayFromWhere(where: Where) {
+  let display = "";
+  const propertyName = getDisplayFromNodeRef(where.nodeRef) ?? getNameFromRef(where);
+  if (!propertyDropList.includes(propertyName)) display += propertyName;
+  if (where.in) display += " " + getDisplayFromList(where, true);
+  if (where.notIn) display += getDisplayFromList(where, false);
+  if (where.operator) display = getDisplayFromOperator(propertyName, where);
+  if (where.range) display = getDisplayFromRange(propertyName, where);
+  if (where.null) display += " is null";
+  return display;
+}
+
+export function getDisplayFromOrderByList(orderByList: OrderLimit[]) {
+  let display = " the ";
+  if (orderByList.length === 1) display += getDisplayFromOrderBy(orderByList[0]);
+  else {
+    display += "[";
+    for (const [index, orderBy] of orderByList.entries()) {
+      display += getDisplayFromOrderBy(orderBy);
+      if (index !== orderByList.length - 1) display += ",";
+    }
+    display += "]";
+  }
+
+  return display + " of the following ";
+}
+
+export function getDisplayFromOrderBy(orderBy: OrderLimit) {
+  let display = "";
+  if (orderBy.variable) display += orderBy.variable + ".";
+  const propertyName = getNameFromRef(orderBy);
+  if (!propertyDropList.includes(propertyName)) display += propertyName + " ";
+  if (orderBy.limit === 1) {
+    if ("descending" === orderBy.direction) display = "latest" + display;
+    if ("ascending" === orderBy.direction) display = "earliest" + display;
+  } else if (orderBy.direction) display = orderBy.direction + " " + display;
+  return display;
+}
+
+export function getDisplayFromLogic(title: string) {
+  switch (title) {
+    case "exclude":
+      return "<span style='color: red;'>exclude</span> ";
+    case "or":
+      return "<span style='color: blue;'>or</span> ";
+    case "and":
+      return "<span style='color: orange;'>and</span> ";
+    default:
+      return "<span style='color: orange;'>and</span> ";
+  }
+}
+
+export function getDisplayFromRange(propertyName: string, where: Where) {
+  const property = " " + propertyName;
+  let display = property + " between ";
+  display += where.range.from.value + " and " + where.range.to.value + " " + where.range.to.unit;
+  return display;
+}
+
+export function getDisplayFromOperator(property: string, where: Where) {
+  let display = "";
+
+  if (property.toLowerCase().includes("date")) {
+    display += getDisplayFromDateComparison(where);
+  } else {
+    if (where.variable) display += where.variable + ".";
+    display += property + " ";
+    display += where.operator + " ";
+    if (where.relativeTo) {
+      let relativeTo = "";
+      if (where.relativeTo.parameter) relativeTo += where.relativeTo.parameter;
+      if (relativeTo) relativeTo += ".";
```

Comment:
Just wondering if the previous 3 lines can be simplified.  parameter is only added if its assigned, and same goes for the ".".
```
      let relativeTo = "";
      if (where.relativeTo.parameter) relativeTo += where.relativeTo.parameter;
      if (relativeTo) relativeTo += ".";
```

so something like...

```
    let relativeTo = where.relativeTo.parameter ? (where.relativeTo.parameter + ".")  : "";
```